### DISCUSSION TITLE
Is there a ~`type Replace<T, R extends Partial<Record<keyof T, any>>> = Omit<T, keyof R> & R;`

### DISCUSSION THREAD

**Discussion by ftzi** [millsp/ts-toolbelt#266] [Q&A]
What I want to to is to for example safely replace {c: number} to {c: string}. The `type Replace<T, R extends Partial<Record<keyof T, any>>> = Omit<T, keyof R> & R;` I have allows exactly this, but I couldn't find a type in this lib that does this, **by having R extending T**, so it helps me building R and keeps it type safe.

There are similar types in the lib, like

`Update<O extends object, K extends Key ...`
`Modify<O extends object, OMod extends object>`
`type Overwrite<O extends object, O1 extends object>`

But none of them, that I've found, have a second generic that extends the origin obj type.

**Comment by tonivj5** [comment]
This time there's no lucky 😆 I think there isn't exactly that type. `Overwrite` https://github.com/millsp/ts-toolbelt/blob/master/sources/Object/Overwrite.ts is the most approximated type, it only replaces with the props of `R` which `T` has.

By the way, I think it's a useful feature and it could be added, or via a third generic argument as a "configuration" to `Overwrite` (e.g, like `deep` or `match` in other types)
